src/peft/tuners/tuners_utils.py

```diff
@@ -398,6 +398,26 @@ def _check_merge_allowed(self):
                 + example_code
             )
 
+    def _check_target_module_compatiblity(self, peft_config: PeftConfig, model: nn.Module, target_name: str):
+        """
+        Prevent applying LoRA to forbidden modules in specific architectures (e.g., Mamba).
+        """
+        lora_like_types = {"LORA", "ADALORA", "XLORA", "RANDLORA"}
+        forbidden_modules = {"out_proj", "conv1d"}
+        mamba_model_types = {"falcon_h1", "mamba", "mamba2", "falcon_mamba"}
+
+        if (
+            peft_config.peft_type in lora_like_types
+            and hasattr(model, "config")
+            and getattr(model.config, "model_type", None) in mamba_model_types
+        ):
+            if any(mod in forbidden_modules for mod in peft_config.target_modules):
+                raise ValueError(
+                    f"[PEFT:{peft_config.peft_type}] target_modules {peft_config.target_modules} contain forbidden modules "
+                    f"for Mamba-based models (model_type={model.config.model_type}): {forbidden_modules}. "
+                    "Please exclude them to avoid compatibility issues."
+                )
```

Comment:
Let's move this function body into a standalone function (no need for it to be a method), something like `_check_target_modules_mamba` or so. Then `_check_target_module_compatiblity` will just call `_check_target_modules_mamba`. IMO, this improves clarity and makes it easier to extend in the future.